streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -487,7 +438,14 @@ public void punctuate(final ProcessorNode node, final long timestamp, final Punc
         }
 
         try {
-            node.punctuate(timestamp, punctuator);
+            StreamsMetricsImpl.maybeMeasureLatency(
+                () -> {
```

Comment:
Are we ever going to need to use the return value here? Instead of using a `Supplier` maybe we could use a `Runnable` in the signature and we could get rid of the `return null` statements.

Reply:
+1. I think the appropriate functional interface is ‘Java.util.function.Consumer’